modules/services/twmn.nix

```
@@ -0,0 +1,347 @@
+          default = { };
+          example = {
+            curve = 19;
+            duration = 618;
+          };
+          description = ''
+            options for the notification disappearance's animation.
+          '';
+        };
+        bounce = {
+          enable = mkEnableOption ''
+            notification bounce when displaying next notification directly.
+          '';
+          duration = mkOption {
+            type = types.ints.unsigned;
+            default = 500;
+            example = 618;
+            description = "the bounce animation duration in milliseconds.";
+          };
+        };
+      };
+      color = mkOption {
+        type = types.str;
+        default = "#000000";
+        example = "lightgray";
+        description = ''
+          notification's background color, RGB hex and keywords (e.g.
+          lightgray) are supported.
+        '';
+      };
+      height = mkOption {
+        type = types.ints.unsigned;
+        default = 18;
+        example = 42;
+        description = ''
+          Height of the slide bar. Useful to match your tiling window
+          manager's bar.
+        '';
+      };
+      offset = {
+        x = mkOption {
+          type = types.int;
+          default = 0;
+          example = 50;
+          description = ''
+            offset of the notification' slide starting point in pixels on the
+            horizontal axis (positive is rightward).
+          '';
+        };
+        y = mkOption {
+          type = types.int;
+          default = 0;
+          example = -100;
+          description = ''
+            offset of the notification' slide starting point in pixels on the
+            vertical axis (positive is upward).
+          '';
+        };
+      };
+      opacity = mkOption {
+        type = types.ints.between 0 100;
+        default = 100;
+        example = 80;
+        description = "the notification window's opacity.";
+      };
+      position = mkOption {
+        type = types.enum [
+          "tr"
+          "top_right"
+          "tl"
+          "top_left"
+          "br"
+          "bottom_right"
+          "bl"
+          "bottom_left"
+          "tc"
+          "top_center"
+          "bc"
+          "bottom_center"
+          "c"
+          "center"
+        ];
+        default = "top_right";
+        example = "bottom_left";
+        description = ''
+          position of the notification slide. The notification will slide
+          in vertically from the border if placed in
+          <literal>top_center</literal> or <literal>bottom_center</literal>,
+          horizontally otherwise.
+        '';
+      };
+    };
+  };
+
+  #################
+  # Implementation
+
+  config = mkIf cfg.enable {
+    home.packages =
+      lib.optional (!isNull cfg.text.font.package) cfg.text.font.package
+      ++ [ pkgs.twmn ];
+
+    home.file."twmn.conf" = {
+      target = ".config/twmn/twmn.conf";
```

please use `xdg.configFile` here

```suggestion
    xdg.configFile."twmn/twmn.conf".text = let
```